craft_parts/utils/snap_utils.py

```diff
@@ -0,0 +1,87 @@
+#
+# Copyright 2023 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Utility functions for snaps."""
+
+import os
+import shutil
+from typing import Optional
+
+import craft_parts.errors
+
+
+def _find_command_path_in_root(root: str, command_name: str) -> Optional[str]:
+    """Find the path of a command in a given root path."""
+    for bin_directory in (
+        "usr/local/sbin",
+        "usr/local/bin",
+        "usr/sbin",
+        "usr/bin",
+        "sbin",
+        "bin",
+    ):
+        path = os.path.join(root, bin_directory, command_name)
+        if os.path.exists(path):
+            return path
+
+    return None
+
+
+def get_host_command(command_name: str) -> str:
+    """Return the full path of the given host tool.
+
+    :param command_name: the name of the command to resolve a path for.
+    :return: Path to command
+
+    :raises SnapcraftError: if command_name was not found.
+    """
+    tool = shutil.which(command_name)
+    if not tool:
+        raise craft_parts.errors.CommandNotFoundError(
+            f"A tool craft-parts depends on could not be found: {command_name!r}",
+            resolution="Ensure the tool is installed and available, and try again.",
+        )
+    return tool
+
+
+def get_snap_command_path(command_name: str) -> str:
```

Comment:
Please put in some tests for this. I'd recommend using `monkeypatch.setenv` to change the environment variables